### PR TITLE
Add PCMSolver

### PR DESCRIPTION
**Pull request by loriab**
PCMSolver is a quantum chemistry library with parts in C++, Fortran, and Python. Pretty much this recipe is RTG, but a few things I need to address before requesting review.
- [x] package v1.2.x or v1.3.x ? get my cmake/windows changes into upstream. Ans: from primary developer's response below, we're going to package v1.2.x for c-f and then update to v1.3.1 once that release gets finalized
- [x] see if RDR wants to be maintainer. Ans: yes (see below)
- [x] release the constraint on one Py that's lowering build times for iteration
- [x] decide whether to package exe or just copy it for tests section. Ans: packaging exe until there's a reason not to
- [x] only W Fortran compiler I could find that worked was ming. will I be able to link Psi4 built with clang-cl with this? Ans: history with dkh in #22328 that's also Fortran inspires confidence that PCMSolver built with ming and with GNUtoMS will work with Psi4.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- ~Source is from official source.~ (Yes, approximately. It's the latest in the last release series plus some Windows, syntax, and cmake modernizations https://github.com/PCMSolver/pcmsolver/pull/203)
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- ~If static libraries are linked in, the license of the static library is packaged.~ (Static libs are purged)
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by robertodr** [comment]
I'm OK with being a maintainer. If not too much work later on, I'd say it's better to first package 1.2.3 and then let the bots update for 1.3.x (once 1.3.1 is minted)

**Comment by loriab** [comment]
https://github.com/orgs/conda-forge/teams/help-c-cpp, ready for review!

**Comment by loriab** [comment]
@conda-forge/help-c-cpp, ready for review!

File: recipes/pcmsolver/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    - python %PREFIX%\\Library\\bin\\go_pcm.py --inp molecule.inp --exe %PREFIX%\\Library\\bin  # [win]
+    - cat molecule.out                                             # [unix]
+    - type molecule.out                                            # [win]
+
+about:
+  home: https://github.com/PCMSolver/pcmsolver
+  dev_url: https://github.com/PCMSolver/pcmsolver
+  doc_url: https://pcmsolver.readthedocs.io/en/stable/
+  doc_source_url: https://github.com/PCMSolver/pcmsolver/tree/master/doc
+  license: LGPL-3.0-only
```
[the remark this answers is not in the document]
Comment (loriab) [reply recipes/pcmsolver/meta.yaml:85]: LGPL3 + MIT + MIT-0 should cover it.

**Comment by loriab** [comment]
> It looks like your python module is pure python. If so, Could you please separate the python module into a noarch: python output so that it is built once instead of for each python version? I'm not exactly sure of the dependency relationship between your various outputs, but you would have two or three outputs: pcmsolver, libpcm, pcmsolver-python(?).

Ordinarily I'm all for multiple outputs so the user can choose the base library or the language bindings. This project's an odd case where the python files are parsing helpers for the standard input mode of the compiled library (https://pcmsolver.readthedocs.io/en/stable/users/input.html#input-description). So, by my understanding, the full library plus py helpers are the only package that's useful to users. I could still split them up with a dummy (=user never directly installs) py package, but I think that's going to be hard to bring about cleanly and still do the main ctest testing in build.sh/bld.bat that itself needs Python (https://github.com/PCMSolver/pcmsolver/blob/master/tests/input/input_cpcm.cpp#L42-L48). I understand the CI time minimization motivation, but this project is in maintenance mode (last release in 2020 https://github.com/PCMSolver/pcmsolver/tags), so it's not going to be eating up resources. I don't like to push back on such a reasonable request, but I don't think this is a great candidate for a split-output recipe. What do you think?

**Comment by carterbox** [comment]
Looks like the python scripts are pre-processors/parsers for the `run_pcm` binary. Why would users of the C/Fortran API who link dynamically to `libpcm.so.1` need the python scripts?

We can also ask conda to do the splitting if the build system isn't set up to break the installation into multiple steps. I've added a commit which demonstrates this kind of syntax. You will need to move some of the existence tests around and add platform selectors so that libpcm is correctly split on Windows.

You'll also need to use the canonical repository for the source download not your personal one. Conda recipes can apply git patches if needed.

**Comment by loriab** [comment]
> Looks like the python scripts are pre-processors/parsers for the run_pcm binary. Why would users of the C/Fortran API who link dynamically to libpcm.so.1 need the python scripts?

It's actually the run_pcm binary that's optional, and the python scripts are the parsers for the library-without-binary. As far as I know, one uses the API for the overall commands and the Python for the options parsing. I suspect this came about so the project could service multiple downstream packages that were in different languages and _very_ set in their ways. But I shouldn't be troubling you with all that. I'll fix it up so it looks more canonical.

> We can also ask conda to do the splitting if the build system isn't set up to break the installation into multiple steps. I've added a commit which demonstrates this kind of syntax. You will need to move some of the existence tests around and add platform selectors so that libpcm is correctly split on Windows.

Will do. Your proposal is certainly the more canonical packaging arrangement.

> You'll also need to use the canonical repository for the source download not your personal one. Conda recipes can apply git patches if needed.

Ok, I'll drop it all into a patch. I think the upstream dev / co-recipe-maintainer will mint a v1.2.4 once the buildsystem changes settle down.

Thanks for all your comments!

**Comment by loriab** [comment]
I think your concerns are addressed, @carterbox, and this is ready for consideration again. No particular urgency on this PR. I'm curious to see how a noarch generates here, as each arch has a different build string for the py package.

**Comment by carterbox** [comment]
@conda-forge/help-c-cpp, please review

**Comment by loriab** [comment]
In case you, too, are worried about selectors in a noarch like https://github.com/conda-forge/staged-recipes/pull/22286/files#diff-793925cb8d7cb1c1f77b1a259178a638e058b7f01abf7a1b60d91127c8994b1eR41-R54 (the windows tests don't work without them), I can see a few strategies:
1. Adopt the semi-noarch approach characterized by `__win`  and described [here](https://github.com/conda-forge/conda-forge.github.io/pull/1839/files#diff-09a3e096a4c36c98cbeb1dbefbed10244ef98032bafee52ece10d9e41bf3e922R1041). My concern with that is it's designed for arch-dependent `dependency packages`, not arch-dependent file positions (like the python script in bin/).
2. Drop the noarch but separate the py packages into a per-arch per-py setup that only copies files. Model: https://github.com/conda-forge/libxc-feedstock/blob/main/recipe/meta.yaml#L138-L140 . That would involve 3 expensive library builds (one per arch) and 12 cheap py copies (one per arch per py).
3. Merge the "noarch" as-is, and hope my worries are mistaken.
4. Revert to [3224aef](https://github.com/conda-forge/staged-recipes/pull/22286/commits/3224aefdc1a016178baaa5b220185c35f0188e2e) where pcmsolver is monolithic and wasteful of CI but self-contained.

I'm glad to do however you advise, @carterbox . Sorry this package is so peculiar.

**Comment by carterbox** [comment]
Because the recipe has arch-specific outputs (libpcm), the noarch: python package (pcmsolver) will be built on all platforms. If the package hashes collide, then there will be only pcmssolver variant available for download. This is fine if the noarch packages are being built correctly because they will be valid installs on all platforms. We can download the artifacts from the builds on azure to check if this is the case.

If the packages are different, then we go with Option 1 (one pcmsolver for each platform shared between many python versions).

**Comment by carterbox** [comment]
The contents of the noarch package generated by Linux seems correct. Let's just check that the package hashes are the same for all platforms from the logs.

**Comment by carterbox** [comment]
@loriab, I'm pretty sure one noarch package will work for all platforms. If you have problems, revert to Option 1 by removing the custom build string and adding

```
      - __linux  # [linux]
      - __osx    # [osx]
      - __win    # [win]
```
To the run requirements.

**Comment by loriab** [comment]
Sorry I missed your recent messages, @carterbox. Thanks for sharing the info on noarch strategies. I'll test pcmsolver with psi4 to confirm the unified py build string is acting as it ought and, if not, switch to the `__win` strategy you suggest. Thanks for the review and merge.

**Pull request merged**